src/context/WalletProvider.tsx

```diff
@@ -511,10 +516,34 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
         throw err
       }
     } else {
-      // EVM tx
+      // EVM txs
       const signer = state.provider!.getSigner()
 
       try {
+        // EVM + Gnosis Safe tx
+        if (state.name === WALLET_NAMES.WalletConnect && state.subName === WALLET_SUBNAME.GnosisSafe && state.walletProvider instanceof WalletConnectProvider) {
+          /*
+            Gnosis Safe cannot immediately return the transaction by design.
+            The multisig signature will be made later.
+            It remains only to wait for the appearance of a new transaction from the sender's address.
+          */
+
+          // no await - no response expected
+          signer?.sendTransaction(transaction)
+
+          const waitForGnosisSafeTxHash: () => Promise<string> = () => {
```

Comment:
it would be better to move from main code

Reply:
Ok, moved